htdocs/compta/facture/class/facture.class.php

```diff
@@ -1730,14 +1730,17 @@ public function fetch($rowid, $ref = '', $ref_ext = '', $notused = '', $fetch_si
 					$this->error = $this->db->error();
 					return -3;
 				}
+
+				$this->db->free($resql);
+
 				return 1;
 			} else {
-				$this->error = 'Invoice with id='.$rowid.' or ref='.$ref.' or ref_ext='.$ref_ext.' not found';
```

Comment:
Why removing the set of ->error. It is required to return the reason of the fetch not found to the caller.

Reply:
@eldy
I don't understand why to keep an error in this class if it's not considered like an error (returning 0).
I think it's in the view (card) you show or not a message with a forbidden access.

Reply:
Only > 0 is a success
If you have an id and want to fetch it, and you do not found it. This is not the expected behaviour. It is not a technical "error" (< 0) but error is not just technical error, it is not expected result.